api/Insight_API_documentation.md

```diff
@@ -0,0 +1,1200 @@
+| M              | `int64`      |   (optional) Address end **REVIEW/SUGGEST ALTERNATIVE**       |  
+| noTxList              | `boolean`      |   (optional) If `noTxList` = '0', response includes a `noTxList`. If `noTxList` = '1', response will not include a `noTxList`      |  
+
+**Request Example:**
+
+```GET /addr/DshRLMBrqyDmMpp9gcz3bLgCnjMa75mrHsS?from=100&to=200?noTxList=1 ```
+
+**Request Response:**
+
+```
+{
+    "addrStr": "DshRLMBrqyDmMpp9gcz3bLgCnjMa75mrHsS",
+    "balance": 0,
+    "balanceSat": 0,
+    "totalReceived": 93.00360718,
+    "totalReceivedSat": 9300360718,
+    "totalSent": 93.00360718,
+    "totalSentSat": 9300360718,
+    "unconfirmedBalance": 0,
+    "unconfirmedBalanceSat": 0,
+    "unconfirmedTxApperances": 0,
+    "txApperances": 4
+}
+```
+
+<br/>
+
+### /addr/ (balance)
+
+**URL:**  ```GET /addr/{address}/balance```
+
+**Description:** Retrieves address balance.
+
+**Parameters:**
+
+| Parameter           | Type                   |  Description                   | 
+| -------------------- | ---------------------- | ---------------------- | 
+| address              | `string`      |   Address    |  
+
+**Request Example:**
+
+```GET /addr/Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx/balance ```
+
+**Request Response:**
+
+```
+57009706748401
+```
+
+<br/>
+
+### /addr/ (totalSent)
+
+**URL:**  ```GET /addr/{address}/totalSent```
+
+**Description:** Retrieves total amount sent from an address. **REVIEW. Is it from?** 
+
+**Parameters:**
+
+| Parameter           | Type                   |  Description                   | 
+| -------------------- | ---------------------- | ---------------------- | 
+| address              | `string`      |   Address  |  
+
+**Request Example:**
+
+```GET addr/Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx/totalSent ```
+
+**Request Response:**
+
+```
+12384494128495
+```
+
+<br/>
+
+### /addr/ (totalReceived)
+
+**URL:**  ```GET /addr/{address}/totalReceived```
+
+**Description:** Retrieves total amount received by an address.
+
+**Parameters:**
+
+| Parameter           | Type                   |  Description                   | 
+| -------------------- | ---------------------- | ---------------------- | 
+| address              | `string`      |   Address  |  
+
+**Request Example:**
+
+```GET /addr/Dcur2mcGjmENx4DhNqDctW5wJCVyT3Qeqkx/totalReceived ```
+
+**Request Response:**
+
+```
+69394400235357
+```
+
+<br/>
+
+### /addr/ (unconfirmedBalance)
+
+**URL:**  ```GET /addr/{address}/unconfirmedBalance```
+
+**Description:** Retrieves unconfirmed balance for an address.
+
+**Parameters:**
+
+| Parameter           | Type                   |  Description                   | 
+| -------------------- | ---------------------- | ---------------------- | 
+| address              | `string`      |   Address |  
+
+**Request Example:**
+
+```GET /addr/DseRvgwcGeCJJC4AUWimHSj8YzRMJ4ryvBM/unconfirmedBalance```
+
+**Request Response:**
+
+```
+0
```

Comment:
Just went with `581`, as it would take me some time pinging the API to find a non-zero integer. Feel free to suggest another number if this one doesn't make sense.

Reply:
That's pretty microscopic, but ok.  You can check the mempool page on the explorer for unconfirmed transactions, but since they are transient results, it might as well be random.

Reply:
Went with 1581, as live block explorer is taking forever to sync (assume this is expected),
![image](https://user-images.githubusercontent.com/1634777/46635368-042cf700-cb09-11e8-8ad9-507fcff3418f.png)